sympy/simplify/hyperexpand.py

```diff
@@ -1661,14 +1661,14 @@ def try_shifted_sum(func, z):
     abuckets, bbuckets = sift(func.ap, _mod1), sift(func.bq, _mod1)
     if len(abuckets[S.Zero]) != 1:
         return None
-    r = abuckets[S.Zero][0]
+    r = int(round(abuckets[S.Zero][0]))
     if r <= 0:
         return None
     if S.Zero not in bbuckets:
         return None
     l = list(bbuckets[S.Zero])
     l.sort()
-    k = l[0]
+    k = int(round(l[0]))
```

Comment:
`nbp.remove(k)` appears below -- isn't it possible that `k` will not be there? or are you just getting ints as floats (in which case `round` is not necessary?

Reply:
It also does not appear that sort is necessary -- can we just use `k = int(round(min(bbuckets[S.Zero])))` instead?

Reply:
> or are you just getting ints as floats (in which case `round` is not necessary?
Yeah, it should `Int` but I'm getting a `Float`.

Reply:
> It also does not appear that sort is necessary -- can we just use `k = int(round(min(bbuckets[S.Zero])))` instead?

Yeah, seems like we can.

Reply:
> `nbp.remove(k)` appears below -- isn't it possible that `k` will not be there? 

I don't think it is possible that `k` will not be there.

> or are you just getting ints as floats (in which case `round` is not necessary?

Yeah, it should `Int` but I'm getting a `Float`.

Reply:
If you don't use `round` then an error will be raised if k is not present during the `remove` attempt. You could put an inline note at that line saying that if this raises then we have to look at whether round should be used to calculate k and the raw k kept for removal purposes.

Reply:
What is the value of k before this line?